app/ts/PreferencesTab/Preferences/PartialValueFunctions/PartialValueFunctionUtil.ts

```diff
@@ -21,53 +23,46 @@ function getXValues(pvf: IPvf, usePercentage: boolean): ['x', ...number[]] {
   ];
 }
 
-export function getPercentifiedNumber(
-  value: number,
-  usePercentage: boolean
-): number {
-  if (usePercentage) {
-    return significantDigits(value * 100);
-  } else {
-    return significantDigits(value);
-  }
-}
-
-function intermediateX(pvf: IPvf, usePercentage: boolean): number[] {
-  if (pvf.cutoffs) {
+function intermediateX(pvf: TPvf, usePercentage: boolean): number[] {
+  if (isPieceWiseLinearPvf(pvf)) {
     return _.map(pvf.cutoffs, (value: number) => {
-      return getPercentifiedNumber(value, usePercentage);
+      return getPercentifiedValue(value, usePercentage);
     });
   } else {
     return [];
   }
 }
 
 function getYValues(
-  pvf: IPvf,
+  pvf: TPvf,
   criterionTitle: string
 ): [string, 1, ...number[]] {
   return [criterionTitle, 1, ...intermediateY(pvf), 0];
 }
 
-function intermediateY(pvf: IPvf): number[] {
-  return pvf.values ? pvf.values : [];
+function intermediateY(pvf: TPvf): number[] {
+  if (isPieceWiseLinearPvf(pvf)) {
+    return pvf.values;
+  } else {
+    return [];
+  }
 }
 
-export function getBest(pvf: IPvf, usePercentage: boolean): number {
+export function getBest(pvf: TPvf, usePercentage: boolean): number {
```

Comment:
Throw if no 'direction'?

Reply:
if we pass `TPvf` in the direction should already be set right? so the issue should happen earlier in the chain